File: android/app/src/main/java/com/smartenergy/tracker/network/ApiService.java
package com.smartenergy.tracker.network;

import com.google.gson.JsonObject;
import com.smartenergy.tracker.model.AlertItem;
import com.smartenergy.tracker.model.Appliance;

import java.util.List;
import java.util.Map;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface ApiService {

    @GET("api/appliances")
    Call<JsonObject> getAppliances();

    @POST("api/appliances/{id}/toggle")
    Call<JsonObject> toggleAppliance(@Path("id") String id, @Body Map<String, Boolean> body);

    @POST("api/appliances/{id}/anomaly")
    Call<JsonObject> testAnomaly(@Path("id") String id, @Body Map<String, Boolean> body);

    @GET("api/alerts")
    Call<JsonObject> getAlerts(@Query("limit") int limit);

    @POST("api/alerts/{id}/resolve")
    Call<JsonObject> resolveAlert(@Path("id") long id);

    @GET("api/analytics/attribution")
    Call<JsonObject> getAttribution();

    @GET("api/analytics/historical")
    Call<JsonObject> getHistorical(@Query("range") String range);
}

File: android/app/src/main/java/com/smartenergy/tracker/network/ApiService.kt
package com.smartenergy.tracker.network

import com.smartenergy.tracker.model.*
import retrofit2.Response
import retrofit2.http.*

interface ApiService {
    @GET("api/appliances")
    suspend fun getAppliances(): Response<ApiResponse<List<Appliance>>>

    @POST("api/appliances/{id}/toggle")
    suspend fun toggleAppliance(
        @Path("id") id: String,
        @Body body: Map<String, Boolean>
    ): Response<ToggleResponse>

    @POST("api/appliances/{id}/anomaly")
    suspend fun injectAnomaly(
        @Path("id") id: String,
        @Body body: Map<String, Any>
    ): Response<ApiResponse<Any>>

    @GET("api/alerts")
    suspend fun getAlerts(): Response<ApiResponse<List<AlertItem>>>

    @POST("api/alerts/{id}/resolve")
    suspend fun resolveAlert(@Path("id") id: Long): Response<ApiResponse<Any>>

    @GET("api/schedules/scenes")
    suspend fun getScenes(): Response<ApiResponse<List<SceneItem>>>

    @POST("api/schedules/scenes/{id}/apply")
    suspend fun applyScene(@Path("id") id: String): Response<ApiResponse<Any>>

    @GET("api/schedules/rules")
    suspend fun getRules(): Response<ApiResponse<List<RuleItem>>>

    @GET("api/schedules/load-shifting")
    suspend fun getLoadShifting(): Response<ApiResponse<LoadShiftingData>>

    @GET("api/analytics/forecast")
    suspend fun getForecast(): Response<ApiResponse<ForecastData>>

    @POST("api/simulation/scenario")
    suspend fun triggerScenario(@Body body: Map<String, String>): Response<ApiResponse<Any>>

    @POST("api/simulation/reset")
    suspend fun resetSimulation(): Response<ApiResponse<Any>>
}
